File: app/src/main/java/com/cargopull/executor_driver/interactor/CommonGateway.java
package com.cargopull.executor_driver.interactor;

import androidx.annotation.NonNull;
import io.reactivex.Flowable;

/**
 * Общий гейтвей получения любых данных потоком.
 */
public interface CommonGateway<D> {

  /**
   * Ожидает данные.
   *
   * @return {@link Flowable<D>} данные.
   */
  @NonNull
  Flowable<D> getData();
}

File: app/src/main/java/com/cargopull/executor_driver/interactor/CommonGateway.kt
package com.cargopull.executor_driver.interactor

import io.reactivex.Flowable

/**
 * Общий гейтвей получения любых данных потоком.
 */
interface CommonGateway<D> {

    /**
     * Ожидает данные.
     *
     * @return [<] данные.
     */
    val data: Flowable<D>
}
